source/sessions/tests/unified/snapshot-sessions.json

```diff
@@ -0,0 +1,926 @@
+{
+  "description": "snapshot-sessions",
+  "schemaVersion": "1.0",
+  "runOnRequirements": [
+    {
+      "minServerVersion": "5.0",
+      "topologies": [
+        "replicaset",
+        "sharded-replicaset"
+      ]
+    }
+  ],
+  "createEntities": [
+    {
+      "client": {
+        "id": "client0",
+        "observeEvents": [
+          "commandStartedEvent"
+        ],
+        "ignoreCommandMonitoringEvents": [
+          "findAndModify",
+          "insert",
+          "update"
+        ]
+      }
+    },
+    {
+      "database": {
+        "id": "database0",
+        "client": "client0",
+        "databaseName": "database0"
+      }
+    },
+    {
+      "collection": {
+        "id": "collection0",
+        "database": "database0",
+        "collectionName": "collection0",
+        "collectionOptions": {
+          "writeConcern": {
+            "w": "majority"
+          }
+        }
+      }
+    },
+    {
+      "session": {
+        "id": "session0",
+        "client": "client0",
+        "sessionOptions": {
+          "isSnapshot": true
```

Comment:
Can we specify `"causalConsistency": false` here and on `session1`? `causalConsistency` is [true by default ](https://github.com/mongodb/specifications/blob/master/source/causal-consistency/causal-consistency.rst#causalconsistency) which will cause an error with `"isSnapshot": true`.

Reply:
The default `causalConsistency` value is "not supplied". So in that case it should be translated to `!isSnapshot` (now it's translated to `true`)?
I think drivers should account for this case, as we don't want mandate users to always  specify `"causalConsistency": false` if snapshot session is requested.

Reply:
I agree that specifying `causalConsistency = false` just for `isSnapshot = true` should not be a burden on the users. 

But if a user (or a test) sets `isSnapshot` to true, drivers should set `causalConsistency` to false? I suppose I'm confused about when drivers should explicitly _error_ because of simultaneously true `isSnapshot` and `causalConsistency` values, and when drivers should quietly switch one or the other off.

Reply:
You are right, this point is not clear.
[I suggest](https://github.com/mongodb/specifications/pull/1022/files#diff-d87cc9942c0585f49a1d22a6ea656ef51d4a57c27e87b0f147a39890c7617f6eR133-R134) that `causalConsistency`  defaults to true only if no snapshot where requested.